### PR TITLE
Structopt

### PR DESCRIPTION
**Pull request by XBagon**
After a few iterations of trying different ways, I think it's in a somewhat useful form now. Sadly not as advantages as I hoped it would be, but definitely a lot nicer to look at and more structured. For more improvements on top of this we'd need to solve the issues/implement the ideas I opened issues/discussions for (#25, #26, #27).
When doing the #26 refactor, we can then also put the direction of the scale into the `from_str` and use it in the CLI, at the moment this is a regression, as direction now can't be specified. I want to fix this as soon as this is accepted or it's clear it will, so I can then commit on top of this branch (if that makes any sense).
#27 will help with a lot better error messages, that can tell you what exact part of parsing failed.

### CODE REVIEW

File: src/bin/rustmt/chord.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    pub fn execute(self) {
+        println!("Available chords:");
+        for chord in &AVAILABLE_CHORDS {
+            println!(" - {}", chord);
+        }
+    }
+}
+
+#[derive(StructOpt, Debug)]
+#[structopt(alias = "n", about = "Prints out the notes of the <chord>", help = "Examples:\n- C minor\n- Ab augmented major seventh\n- F# dominant seventh / C#\n- C/1", help = "test")]
```
Comment (ozankasikci) [inline src/bin/rustmt/chord.rs:62]: help = "test" ? :)

**Comment by ozankasikci** [comment]
@XBagon It looks more modular than the previous implementation. But there isn't a huge practical benefit as you mentioned, at least not at the moment.

Question: How do we compile the binary after this change? Is it the same as before?

File: src/bin/rustmt/chord.rs (hunk 1 above)
Comment (XBagon) [reply src/bin/rustmt/chord.rs]: Oh, good catch, guess I forgot to remove that.

**Comment by XBagon** [comment]
> Question: How do we compile the binary after this change? Is it the same as before?
Yes I just split the main part into 3, but the core remains the same, so it compiles exactly the same.

**Comment by ozankasikci** [comment]
Thanks for the contribution @XBagon ! 👍 

Merging this.

**Pull request merged**